demo/src/pages/index.js

```diff
@@ -38,10 +41,28 @@ const liveCodeSample = `
 <Button variant={Button.VARIANT.PRIMARY} onClick={() => alert('Hello!')}>Hello!</Button>
 `;
 
+const gdprConsentCookieName = 'yummyCookie'
+
+const options = { expires: 1 /* day */ };
+
 const IndexPage = ({ data }) => {
   const { layout, siteMetadata } = data.site;
   const [searchTerm, setSearchTerm] = useState('');
   const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const [cookieConsent, setCookieConsent] = useState(Cookies.get(gdprConsentCookieName) === 'true'); 
+
+  const previousCookieConsent = usePrevious(cookieConsent);
+
+  const handleCookieConsent = (answer) => {
+    Cookies.set(gdprConsentCookieName, String(!!answer), options);
+    answer && setCookieConsent(true);
+  } 
+
+  useEffect(()=> {
+      if (cookieConsent && previousCookieConsent === false) {
+        alert(`${gdprConsentCookieName} has been approved`)
+      } 
+  }, [cookieConsent, previousCookieConsent])
```

Comment:
You'll notice this is a TON of boilerplate that consumers of the theme would need to concern themselves with. By moving it to the theme, I think we should be pretty opinionated and try and take as much of the pain out of implementing cookie consent as much as possible. To me, that would be: I import the component, render it, and thats it. I don't need to concern myself with setting cookies, dealing with tracking, etc.

I've posted some suggestions on how to achieve this. Ideally this demo file would look like this:

```js
import { CookieConsentDialog } from '@newrelic/gatsby-theme-newrelic'

const HomePage = () => {
  return (
    // ...
    <CookieConstentDialog />
  )
}
```